test/helpers/kubectl.go

```diff
@@ -2296,27 +2304,14 @@ func (kub *Kubectl) ciliumUninstallHelm(filename string, options map[string]stri
 
 // CiliumInstall installs Cilium with the provided Helm options.
 func (kub *Kubectl) CiliumInstall(filename string, options map[string]string) error {
-	var (
-		wg                sync.WaitGroup
-		resourcesToDelete = map[string]string{
-			"configmap":          "cilium-config",
-			"daemonset":          "cilium",
-			"daemonset ":         "cilium-node-init",
-			"deployment":         "cilium-operator",
-			"clusterrolebinding": "cilium cilium-operator",
-			"clusterrole":        "cilium cilium-operator",
-			"serviceaccount":     "cilium cilium-operator",
+	// First try to remove any existing cilium install. This is done by removing resources
+	// from the file we generate cilium install manifest to.
+	if _, err := os.Stat(filename); err == nil {
```

Comment:
This will not work, because we are using new timestamped `cilium.yaml` path for each `CiliumInstall` call. We might either standarize the yaml name and reuse it, or force following a pattern which will make it finding the file with k8s objects to delete easier in this function.

Reply:
Yeah, I have started a discussion around this on our slack channel. It would be great if you can pass on your views. https://cilium.slack.com/archives/C7PE7V806/p1596279201185200?thread_ts=1596279201.185200&cid=C7PE7V806